eva-accession-import/src/main/java/uk/ac/ebi/eva/accession/dbsnp/processors/SubSnpNoHgvsToDbsnpVariantsWrapperProcessor.java

```diff
@@ -85,7 +85,7 @@ private void addSubmittedVariantEntity(SubSnpNoHgvs subSnpNoHgvs,
         DbsnpSubmittedVariantEntity submittedVariantEntity = new DbsnpSubmittedVariantEntity(subSnpNoHgvs.getSsId(),
                                                                                              hash, submittedVariant,
                                                                                              1);
-        submittedVariantEntity.setCreatedDate(getCreatedDate(subSnpNoHgvs));
+        submittedVariantEntity.setCreatedDate(submittedVariant.getCreatedDate());
```

Comment:
This is the same thing as in the other hierarchy. You can add a new argument to the constructor, ideally before `version` to preserve ordering to an extent, then get it from the submitted variant **inside the constructor** instead of via an additional setter call.

This will make sure that any constructed entity is fully correct, instead of depending on an external user to know that the creation date needs to be set afterwards.